### PR TITLE
add '-c r' to engine if engine=='r'

### PR DESCRIPTION
**Pull request by PeterDSteinberg**
anaconda-build was unable to create the engine 'r' without the channel 'r' being included.

### CODE REVIEW

**Comment by srossross** [comment]
I think that this PR looks good. @PeterDSteinberg have you tested this? 

As a side note, if we are hosting r packages, why is this not enabled by default in conda?

**Comment by srossross** [comment]
@PeterDSteinberg I think a better long term solution would be to include a 'channels' section to the `.binstar.yaml` file.

That way we can add channels to the default `.condarc` and the r engine can imply the r channel.

e.g.
```
channels:
    - r
   - defaults
   - sean

engine:
  - r 

script:
   - conda build .
```

**Comment by PeterDSteinberg** [comment]
I agree @srossross  that sounds like a good idea to separate out the channels into separate yaml entry.  I was wondering though if we want to change ```channels``` in your yaml snippet above to ```install_channels``` to better disambiguate from the ```channel``` argument given to ```anaconda build submit```.  In that context ```channel``` controls the user's channel to which the package is uploaded, so it's a bit confusing potentially to see it in two places doing different things.

**Comment by srossross** [comment]
Thank you @PeterDSteinberg, that is a great idea.

**Comment by PeterDSteinberg** [comment]
The latest commit addresses @srossross 's idea of having an ```install_channels``` entry in .binstar.yml.  Now if you have this snippet in your .binstar.yml:
```
install_channels:
    - r
   - defaults
   - sean

engine:
  - r 

script:
   - conda build .
```
then the ```r```, ```defaults```, and ```sean``` channels will be added to build worker's condarc file so that install and build may refer to packages in those channels.

**Comment by PeterDSteinberg** [comment]
The latest 2 commits
 * add a few help notes about ```install_channels``` that shows up in the default .binstar.yml when doing ```anaconda build init``` to start a package metadata.
 * add a test to ensure ```--add channels some_channel``` shows up in build script when ```install_channels``` is in .binstar.yml
 * add a test to ensure the ```r``` ```install_channel``` is added by default to build script when the ```BINSTAR_ENGINE``` is ```r```.

File: binstar_build_client/worker/utils/data/build_script.sh [hunk 1]
```diff
@@ -85,8 +85,9 @@ setup_build(){
 
     echo "export CONDARC=$CONDARC"
     touch "$CONDARC"
-
-    conda config --file "$CONDARC" --add channels defaults
```
Comment (srossross) [inline binstar_build_client/worker/utils/data/build_script.sh]: @PeterDSteinberg is this save to remove?

**Comment by srossross** [comment]
@PeterDSteinberg The code looks good.  Just give me time to test it out and then we should be good to go.

File: binstar_build_client/worker/utils/script_generator.py [hunk 2]
```diff
@@ -176,14 +176,20 @@ def gen_build_script(working_dir, build_data, **context):
 
 
     exports = create_exports(build_data)
-
+    instructions = build_data['build_item_info'].get('instructions', {})
+    install_channels = instructions.get('install_channels', None) or ['defaults']
+    if not 'defaults' in install_channels:
```
Comment (PeterDSteinberg) [inline binstar_build_client/worker/utils/script_generator.py]: @srossross if the user has not given ```defaults``` as an install_channels, it is added automatically to a list of install_channels here.  If we have other logic about automatically adding channels to ```install_channels``` that logic can also go here, like the inclusion of the r channel a couple lines below.

File: binstar_build_client/worker/utils/data/build_script.sh (hunk 1 above)
Comment (PeterDSteinberg) [inline binstar_build_client/worker/utils/data/build_script.sh]: yes safe to remove.  See comment below

**Comment by PeterDSteinberg** [comment]
Looks like this just failed on WIndows 2.7 for an unrelated reason.

**Comment by stephenakearns** [comment]
@srossross - tomorrow (fri) @PeterDSteinberg has some time to work on the team. He's looking at fixing #151 which is related to this one. Is this PR ready to be merged? If not, would be great to have peter work on any feedback you might have.

**Comment by srossross** [comment]
:+1:

**Comment by srossross** [comment]
@PeterDSteinberg can you coordinate with @electronwill to get this new functionality added to the docs?

**Pull request merged**